ci/jenkins/jobs/macros.yaml

```diff
@@ -3,7 +3,7 @@
     builders:
       - shell: |-
           cp /var/lib/jenkins/utils/defaults.yaml ci/jenkins/jobs
-          find ci/jenkins/jobs/ | grep ci/jenkins/jobs/projects | grep -v projects-{jenkins_location}.yaml | xargs rm
+          find ci/jenkins/jobs -name "projects-*" ! -name "projects-{jenkins_location}.yaml" | xargs rm
```

Comment:
find ci/jenkins/jobs/ -type f -name 'projects-*.yaml' ! -name 'projects-{jenkins_location}.yaml' -exec rm {} +

we can prefer to use "-exec rm {} +" instead of "xargs rm" ,
"-exec rm {} +" is built-in to the find command, so we don't need an additional tool like xargs,
 as xargs is tool which might not be available in some minimal installations or restricted environments.

Reply:
We have added the lab Jenkins job updater in jenkins yaml, right? I think this PR is about improving the scripts in the macro yaml. If that's the case, please update the commit message.

Reply:
I fail to associate the change with the description "Modified the the builder used in updater job to enable updater-job for lab Jenkins as well, along with the cloud Jenkins."
The new command has the same effect as the old command if I understand correctly. Do I miss anything?

Reply:
> I fail to associate the change with the description "Modified the the builder used in updater job to enable updater-job for lab Jenkins as well, along with the cloud Jenkins." The new command has the same effect as the old command if I understand correctly. Do I miss anything?

Yes, since we have supported updaters for both lab and cloud Jenkins, this PR is for improving the macro script that @rajnkamr mentioned. Please update both commit message and PR description @jainpulkit22

Reply:
I am somewhat confused about the commit message, I think the only difference in this improvement is it uses find's built-in capabilities to directly filter files safely based on their names. This avoids the overhead of piping all file paths through multiple grep commands, which can slow down the process. You mentioned 
> jenkins_location had to be set to lab

But even with your changes, we still need to set the jenkins_location value.

Reply:
actually i meant that now the jenkins_location that we have to provide is in sync with the jenkins_location for which we are running the updater job, previously what happened if the updater job was for cloud jenkins the the location we have to provide was "lab" and vice-e-versa, so now this discrepancy is also removed.

Reply:
I tried to run these two commands but they have the same result:

> root@a:/root/antrea# find ci/jenkins/jobs/ | grep ci/jenkins/jobs/projects | grep -v projects-lab.yaml
> ci/jenkins/jobs/projects-cloud.yaml

> root@a:/root/antrea# find ci/jenkins/jobs -name "projects-*" ! -name "projects-lab.yaml"
> ci/jenkins/jobs/projects-cloud.yaml